python/ray/data/impl/arrow_block.py

```diff
@@ -188,8 +185,21 @@ def schema(self) -> "pyarrow.lib.Schema":
     def to_pandas(self) -> "pandas.DataFrame":
         return self._table.to_pandas()
 
-    def to_numpy(self) -> np.ndarray:
-        return np.array(self._table)
+    def to_numpy(self, column: str = None) -> np.ndarray:
+        if not column:
+            raise ValueError(
+                "`column` must be specified when calling .to_numpy() "
+                "on Arrow blocks.")
+        if column not in self._table.column_names:
+            raise ValueError(
+                "Cannot find column {}, available columns: {}".format(
+                    column, self._table.column_names))
+        array = self._table[column]
+        if array.num_chunks > 1:
+            # TODO(ekl) combine fails since we can't concat ArrowTensorType?
```

Comment:
Hmm should be easy to add, we can look at supporting this.

Reply:
Ok, do you have any pointers? I can give it a try